src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -459,6 +459,30 @@ CHIP_ERROR DiscoveryImplPlatform::FinalizeServiceUpdate()
     return ChipDnssdFinalizeServiceUpdate();
 }
 
+void DiscoveryImplPlatform::RegisterResolverDelegate(ResolverDelegate * delegate)
+{
+    for (auto & resolverDelegate : mResolverDelegates)
+    {
+        if (resolverDelegate == nullptr || resolverDelegate == delegate)
```

Comment:
If the assumption is that the same delegate can be registered twice, the loop shouldn't break when `resolverDelegate == nullptr` as the delegate can be stored further.

Reply:
The implementation assumes that `RegisterResolverDelegate` can be called twice with the same argument (otherwise, the `resolverDelegate == delegate` check wouldn't be needed. If so, then in case `resolverDelegate == nullptr`, then we should continue the loop and try to check if there's no entry equal to `delegate`. That's my point.

Reply:
Safety seems reasonable here, marking as resolved.